src/main/java/org/craftercms/studio/impl/v2/upgrade/operations/BlueprintRenameFolderUpgradeOperation.java

```diff
@@ -0,0 +1,136 @@
+import org.craftercms.studio.api.v1.service.security.SecurityService;
+import org.craftercms.studio.api.v2.exception.UpgradeException;
+import org.craftercms.studio.api.v2.service.security.internal.UserServiceInternal;
+import org.craftercms.studio.impl.v1.repository.git.GitContentRepositoryHelper;
+import org.eclipse.jgit.api.Git;
+import org.eclipse.jgit.api.Status;
+import org.eclipse.jgit.api.errors.GitAPIException;
+import org.eclipse.jgit.lib.Repository;
+import org.springframework.beans.factory.annotation.Required;
+import org.springframework.web.context.ServletContextAware;
+
+import javax.servlet.ServletContext;
+import java.io.File;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import static org.craftercms.studio.api.v1.util.StudioConfiguration.BLUE_PRINTS_PATH;
+import static org.craftercms.studio.api.v1.util.StudioConfiguration.REPO_BLUEPRINTS_UPDATED_COMMIT_MESSAGE;
+import static org.craftercms.studio.impl.v1.repository.git.GitContentRepositoryConstants.GIT_COMMIT_ALL_ITEMS;
+
+/**
+ * Implementation of {@link org.craftercms.studio.api.v2.upgrade.UpgradeOperation} that renames the blueprint in the
+ * global repository.
+ * @author Dejan Brkic
+ */
+public class BlueprintRenameFolderUpgradeOperation extends AbstractUpgradeOperation implements ServletContextAware {
+
+    private static final Logger logger = LoggerFactory.getLogger(BlueprintRenameFolderUpgradeOperation.class);
+
+
+    public static final String CONFIG_KEY_BLUEPRINT_FOLDER_NAME = "blueprintFolderName";
+    public static final String CONFIG_KEY_NEW_FOLDER_NAME = "newFolderName";
+
+    protected ServletContext servletContext;
+    protected ServicesConfig servicesConfig;
+    protected SecurityService securityService;
+    protected UserServiceInternal userServiceInternal;
+
+    protected String blueprintFolderName;
+    protected String newFolderName;
+
+    @Override
+    public void setServletContext(final ServletContext servletContext) {
+        this.servletContext = servletContext;
+    }
+
+    @Required
+    public void setServicesConfig(final ServicesConfig servicesConfig) {
+        this.servicesConfig = servicesConfig;
+    }
+
+    public SecurityService getSecurityService() {
+        return securityService;
+    }
+
+    public void setSecurityService(SecurityService securityService) {
+        this.securityService = securityService;
+    }
+
+
+    public void setUserServiceInternal(UserServiceInternal userServiceInternal) {
+        this.userServiceInternal = userServiceInternal;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void doInit(final Configuration config) {
+        blueprintFolderName = config.getString(CONFIG_KEY_BLUEPRINT_FOLDER_NAME);
+        newFolderName = config.getString(CONFIG_KEY_NEW_FOLDER_NAME);
+    }
+
+    @Override
+    public void execute(final String site) throws UpgradeException {
+        try {
+            GitContentRepositoryHelper helper =
```

Comment:
Maybe we should try to use the inherited `getRepositoryPath(String site)` to:
- Avoid the security & user service dependencies? I remember it caused some issues because when the upgrade runs there is no logged user
- Make sure we can use it for both global & site repos too